src/worker.ts

```diff
@@ -122,7 +124,17 @@ function atomicsWaitLoop (port : MessagePort, sharedBuffer : Int32Array) {
     // Check whether there are new messages by testing whether the current
     // number of requests posted by the parent thread matches the number of
     // requests received.
-    Atomics.wait(sharedBuffer, kRequestCountField, lastSeenRequestCount);
+    
+    if (useAsyncAtomics === true) {
+      // @ts-expect-error
+      const { async, value } =  Atomics.waitAsync(sharedBuffer, kRequestCountField, lastSeenRequestCount);
+
+      return async === true && value;
```

Comment:
Where is this awaited?

Reply:
[Here](https://github.com/piscinajs/piscina/pull/687/files#diff-1245a5bbc4ca0684dff978efcae118b08f62d2afc05812df22d0627985b4273aR88) and [here](https://github.com/piscinajs/piscina/pull/687/files#diff-1245a5bbc4ca0684dff978efcae118b08f62d2afc05812df22d0627985b4273aR155).

They're a IIFE that has a `catch` to handle the errors; it is not awaited explicitly to avoid wrapping a non-promise into a Promise, nevertheless we might be loosing some stack out of the stacktrace. although as it is executed in a single place it might not be that bad but happy to hear your thoughts

Reply:
I still don't see how it's awaited... Maybe to complicated for me to understand

Reply:
Maybe I didn't get what you meant at the first time, but think I might get it now. At a top-level is not really awaited, but more on a fire-and-forget approach, so the Promise is not awaited for resolution but triggered and any error is handled.

This happens on every new message or at a startup

Reply:
That doesn't sound good? Though I don't know enough of the current design to truly know.

Reply:
Yeah, while writing this, I noticed the same; let me see if there's any side-effect on awaiting on it.
I don't see a potential problem as anyways we already await for the task handler to resolve the task, so it should come for free; but let me check and benchmark

Reply:
Refactored into awaiting for the wake up; although the benchmarks requires some re-work, the refactor reflected a nice performance boost